Singularity

```diff
@@ -0,0 +1,103 @@
+  if [ $? -eq 0 ]; then
+      eval "$__conda_setup"
+  else
+      if [ -f "/usr/local/etc/profile.d/conda.sh" ]; then
+          . "/usr/local/etc/profile.d/conda.sh"
+      else
+          export PATH="/usr/local/bin:$PATH"
+      fi
+  fi
+  unset __conda_setup
+
+  conda activate base
+
+%post
+  # Define logging
+  export logpath=/.singularity.d/logs/
+  mkdir -p $logpath
+
+  # ##  Stages are useful when subsequent images are bootstrapped off of an
+  # ##  existing image, e.g. 'base' is stage A, image based on base is B
+  # SINGULARITY_STAGE=A
+  # logno=0
+  # alias logbase='logbase=$(printf $SINGULARITY_STAGE"%02d" $logno)'
+  # alias loginc='logno=$((logno+1))'
+  # ##  logno prints and increments the log number, 'A00', 'A01', etc...
+  # logno_func() {
+  #   echo $SINGULARITY_STAGE$(printf "%02d" $_logno)
+  #   _logno=$((_logno+1))
+  # }
+
+  # alias logno='$(printf $SINGULARITY_STAGE"%02d" $_logno); __l=$(printf $SINGULARITY_STAGE"%02d" $_logno); _logno=$((_logno+1))'
+
+  #  Basic yum dependencies
+  yum update -y | tee $logpath/$logno-yum-update.log
+
+  ##  Use EPEL release for cmake3 and SCL for devtoolset
+  yum install -y \
+    epel-release \
+    centos-release-scl \
+  | tee $logpath/$logno-yum-install-releases.log
+
+  ##  Contains GCC7 and other dev dependencies
+  yum install -y devtoolset-7 | tee $logpath/$logno-yum-install-devtoolset-7.log
+
+  ##  Activate environment in new shell, exit completely at end
+  scl enable devtoolset-7 bash && exit
+
+  ##  Basic dependencies
+  ##  xeyes used for testing and for installing some x11 dependencies
+  yum install -y \
+    wget nano git \
+    cmake3 tbb \
+    xeyes qt5-qtbase libxkbcommon-x11 \
+  | tee $logpath/$logno-yum-install.log
+
+  ln -s /bin/cmake3 /bin/cmake
+
+  ##  gets set by scl, could interfere with miniconda so unset it here
+  unset PYTHONPATH
+
+  #  Install and setup miniconda
+  ##  Download and install minconda
+  curl -sSL http://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o /tmp/miniconda.sh
+  chmod +x /tmp/miniconda.sh
+  /tmp/miniconda.sh -bfp /usr/local/
+  rm -f /tmp/miniconda.sh
+  source /usr/local/bin/activate
+
+  ##  Install python dependencies - defalt to 3.7.5 if no version provided
+  PYTHON_VERSION="${PYTHON_VERSION:-3.7.5}"
+  echo "Building for python version: $PYTHON_VERSION"
+
+  conda install -y python=$PYTHON_VERSION
+  pip install numpy
+
+  mkdir -p /usr/local/src/
+  cd /usr/local/src/
+  git clone https://github.com/European-XFEL/EXtra-foam
+  cd EXtra-foam
+  git checkout $GIT_COMMIT_HASH
+  git log -n 1 | tee -a $logpath/$logno-extra-foam-clone.log
+  git submodule update --init | tee -a $logpath/$logno-extra-foam-submodules.log
+
+  pip install . | tee -a $logpath/$logno-extra-foam-pip-install.log
+
+  ##  Export environment to concretise
+  conda env export -n base -f /.singularity.d/logs/$logno-conda-env-export.log
+  pip freeze > /.singularity.d/logs/$logno-conda-env-export.log
```

Comment:
We can also have `%runscript` section for easy to run `./extra-foam.sif {detector}`.